ipset.go

```diff
@@ -268,59 +137,233 @@ func (s *IPSet) Ranges() []IPRange {
 	}
 	if len(in) > 0 {
 		// Ran out of removals before the end of in.
-		ret = append(ret, in...)
+		min = append(min, in...)
 		if debug {
-			debugf("ret=%v", ret)
+			debugf("min=%v", min)
 		}
 	}
 
-	// TODO: possibly update s.in and s.out, if #110 supports that.
+	s.in = min
+	s.out = nil
+}
 
-	return ret
+// Clone returns a copy of s that shares no memory with s.
+func (s *IPSetBuilder) Clone() *IPSetBuilder {
+	return &IPSetBuilder{
+		in:  append([]IPRange(nil), s.in...),
+		out: append([]IPRange(nil), s.out...),
+	}
+}
+
+// Add adds ip to s.
+func (s *IPSetBuilder) Add(ip IP) { s.AddRange(IPRange{ip, ip}) }
+
+// AddPrefix adds all IPs in p to s.
+func (s *IPSetBuilder) AddPrefix(p IPPrefix) { s.AddRange(p.Range()) }
+
+// AddRange adds r to s. If r is not Valid, AddRange does nothing.
```

Comment:
If on a new line

Also should the other two Add methods above also say the equivalent thing?

Reply:
The other Add methods take types which cannot be invalid, so no such problem applies.